sound/soc/sof/loader.c

```diff
@@ -239,6 +239,10 @@ int snd_sof_load_firmware_raw(struct snd_sof_dev *sdev)
 	if (!fw_filename)
 		return -ENOMEM;
 
+	/* Don't request firmware again if firmware is already requested */
+	if (plat_data->fw)
+		return 0;
```

Comment:
sorry, why do you need to add this?
this would mean the code is called multiple times, and we have allocated fw_filename multiple times as well? Also the state machine would be that sdev->code_loading would be set multiple times, can this really happen?
can you clarify the flow?

Reply:
@plbossart Yes, you are right. snd_sof_load_firmware_raw() will be called multiple times. See [ret = snd_sof_load_firmware(sdev);](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/pm.c#L265) We load firmware in sof_resume(). I think fw_filename is fine since it is allocated by devm_kasprintf()

Reply:
well, at each suspend/resume cycle we allocate a new string, that's not good. You should move the new test before the fw_filename allocation, that way it'll fix two issues :-)

Reply:
@plbossart done